packages/formation-react/src/components/Pagination/Pagination.jsx

```diff
@@ -17,16 +18,30 @@ class Pagination extends Component {
 
   static defaultProps = {
     ariaLabelSuffix: '',
+    disableGATracking: false,
     maxPageListLength: 10,
   };
 
+  onPageSelect = (page, eventID) => {
+    // Propogate via the prop.
+    this.props.onPageSelect(page);
+
+    // Conditionally track the event.
+    if (window.dataLayer && !disableGATracking) {
+      window.dataLayer.push({
+        event: eventID,
+        'paginate-page-number': page,
+      });
+    }
```

Comment:
I'm wondering if this whole chunk could be a function with a default prop to possibly be a more flexible API -

```js
static defaultProps = {
  trackEvent() {
      window.dataLayer.push();
  },
  //...
}

// ...
if (this.props.enableGATracking) this.props.trackEvent();

```

Also wondering if maybe `enableGATracking` should be named agnostic of the analytics library (GA) even though GA is used by default. What are your thoughts?

Reply:
I don't really know of any other analytics library that uses `window.dataLayer` other than GA, so I feel like it has to be GA-specific... it's a good point though! I'll make the update so they can include their own `trackEvent` func prop that defaults to `window.dataLayer.push` 🙂